src/components/Form/Form.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import users from '../../api/users';
+import { Error } from '../Error';
+
+export class Form extends React.Component {
+  state = {
+    userName: '',
+    newTodo: '',
+    errorHidden: true,
+  }
+
+  onChange = (event) => {
+    const { value, name } = event.target;
+
+    this.setState({
+      [name]: value.replace(/[^\w ]/gi, ''),
+    });
+  }
+
+  addTodo = (event) => {
+    const { userName, newTodo } = this.state;
+    const { onAdd, todosLength } = this.props;
+    let newTodoItem;
+
+    event.preventDefault();
+
+    if (userName !== '' && newTodo !== '') {
+      newTodoItem = {
+        id: todosLength + 1,
+        title: newTodo,
+        completed: false,
+        user: users.find(person => person.name === userName),
+      };
+      newTodoItem.userId = newTodoItem.user.id;
+      this.setState({
+        userName: '',
+        newTodo: '',
+      });
+      onAdd(newTodoItem);
+    } else if (userName) {
+      this.setState({
+        error: 'title',
+        errorHidden: false,
+      });
+    } else {
+      this.setState({
+        error: 'user',
+        errorHidden: false,
+      });
+    }
+  }
```

Comment:
too many if else statements, check for empty fields at the beginning of the function